### ISSUE TITLE
Backup marks message in mailbox as read

### ISSUE THREAD

**Issue by GadgetComa** [joeyates/imap-backup#118]
Sorry if I'm missing something here. I searched a lot to see if I overlooked the answer..

imap-backup works well for me (thank you!) except all unread messages in my mailbox get marked as read when the backup runs. Is this expected or is there a config option I'm missing? I'm backing up two iCloud mail accounts.

Thanks!

**Comment by joeyates** [comment]
Hi @GadgetComa 

That's odd.

When imap-backup, unread messages **should** remain flagged as unread. That's certainly the case for the GMail and the other providers I use.

Please re-check that you can replicate this with iCloud. If so, I will activate an iCloud account and see if I can replicate myself.

**Comment by GadgetComa** [comment]
Thanks for the reply. I see this happen every day on two different iCloud accounts. Is there some debug information I can provide to assist in troubleshooting?  Also, if it helps, I can set up a gmail account to test and see if it’s specific to iCloud or something in my environment.

**Comment by GadgetComa** [comment]
It does appear to be iCloud-specific. I just did a backup of a Gmail account using the exact same setup I use for iCloud and the messages are left as unread.

**Comment by joeyates** [comment]
I have now reproduced this on an iCloud account. As this behaviour is limited to iCloud, and the calls made by imap-backup should not change the status of the 'SEEN' flag, I think we can consider this an iCloud bug.

**Comment by joeyates** [comment]
Hi @darkfence thanks for the extra info.

Could you run a quick check for me? Please mark some emails as unread and then re-run imap-backup.
Do the emails get marked as read immediately?

**Comment by darkfence** [comment]
Yes, as soon as the backup starts, unread messages are marked as read.

**Comment by joeyates** [comment]
I've now reproduced this behaviour  on an iCloud account.
Next step is to create a purelymail.com to reproduce it there too.
GMail (and most other providers) do not behave this way.

The "philosophical" question, I suppose, is whether IMAP servers *should* mark emails as read when their content is accessed in any way (as iCloud seems to do), or whether they should expect clients to set the 'Seen' flag explicitly.

I'm looking at a workaround that checks the 'Seen' flag status before and after fetching emails, and removes the 'Seen' flag if it gets set by the operation.

**Comment by joeyates** [comment]
@GadgetComa I've created a branch with a fix for iCloud: `feature/debug-emails-being-marked-as-read`

If you have a change to try it out, that would be very helpful.

You can install it as follows:

```sh
gem uninstall imap-backup # To uninstall the current version
git clone https://github.com/joeyates/imap-backup.git
cd imap-backup/
git checkout --track -b feature/debug-emails-being-marked-as-read origin/feature/debug-emails-being-marked-as-read
gem build imap-backup.gemspec
sudo gem install imap-backup-*.gem
```

@darkfence As soon as I've had a chance to try this out with purelymail.com, I'll let you know!

**Comment by joeyates** [comment]
Hi @darkfence 

I've added the workaround for purelymail.com clients too. If you have a moment please try it out (see comment above).

**Comment by joeyates** [comment]
> @joeyates unfortunately it does not work on my purelymail account. Unread mail was marked as read as previously.

@darkfence Which domain are you using for your purelymail account?

**Comment by darkfence** [comment]
My own domain. Maybe that's the reason...

**Comment by joeyates** [comment]
OK, I'll need to add a configuration option to handle this case.

**Comment by bdillahu** [comment]
Could I suggest (unless its dangerous or something) that this option default to "on"? I've not marked all my email as "read" for all of history, even though it wasn't... My bad for not seeing the switch (or catching the problem in my testing), but just thought having the option be defaulted to "safe" might be better.

**Comment by joeyates** [comment]
@bdillahu Strange you didn't get this setting automatically.

The [default for purelymail is to set the flag to "on"](https://github.com/joeyates/imap-backup/blob/a1e55d16a9024ee94d877dc7a2c76fce3acc8760/lib/imap/backup/email/provider/purelymail.rb#L14).

**Comment by bdillahu** [comment]
Dunno... 

By chance do you look for "imap.purelymail.com" to key off of?

It may have been self inflicted as when I initially set things up I mistyped the "imap" part - then I believe I just edited the config file directly when I realized I had a typo.

**Comment by joeyates** [comment]
That would explain the course of events.

It's hard to imagine how to handle this use case in a system where the config can be manually modified.

Thanks for the clarification! I'll bear this in mind in future and try to find a way to catch this type of problem.

**Comment by bdillahu** [comment]
The only suggestion I can come up with is to maybe check for *purelymail* to handle variations… 

or have the check occur each time you read the config, but then people who particularly want it the other way would have problems, so you probably have the best solution already.
